services/upgrader/errors.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package upgrader
+
+import "fmt"
+
+// InvalidArch indicates that the current operating system or cpu architecture doesn't support upgrades
+type InvalidArch struct{}
+
+func NewInvalidArch() *InvalidArch {
+	return &InvalidArch{}
+}
+
+func (e *InvalidArch) Error() string {
+	return "invalid operating system or processor architecture"
```

Comment:
The red error text we're currently generating isn't concatenated correctly. See image.
<img width="1109" alt="error-incorrect-concatenation" src="https://user-images.githubusercontent.com/57730300/90690965-68519380-e227-11ea-8460-f89e667d380b.png">

Reply:
I going to address all this in the frontend side, just adding the translations in the client instead of using the server messages. This way we no longer need to care about server side generated errors.

Reply:
Finally I decide to fix it in the server side, because fix it in the frontend side have other complications.

Reply:
Now should be fixed.